### PR TITLE
Issue#25: Initial CI setup

### PR DESCRIPTION
**Pull request by alexorlenko**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @omelchenkotimur 
- [ ] @mykytam 
- [ ] @Arthurpanda 
- [ ] @reeztl 
- [ ] @Prosperro-de 
- [ ] @shiftko 

### Second Level Review

- [ ] @DzigMS 
- [ ] @kh0ma 

## Summary of issue

This PR onboards project on GitHub CI

## Summary of change

 Initial CI file "Java CI with Maven" added

## Testing approach

 -

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      - name: Build with Maven
+        run: mvn clean compile install -DskipTests --file pom.xml
```
Comment (kh0ma) [inline .github/workflows/dev.yml]: Why do we need `compile` as it is already included in `install`?

Comment (kh0ma) [inline .github/workflows/dev.yml]: I don't think we need separate tests running from the main build at this point of time. There is no reason for it. Also In the current configuration, you perform a bunch of maven lifecycle steps twice. Let's keep it simple for now.

**Review by Prosperro-de — APPROVED** [review]

**Review by isvyry — APPROVED** [review]

**Review by omelchenkotimur — APPROVED** [review]

File: .github/workflows/dev.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      - name: Build and test with Maven
+        run: mvn clean install --file pom.xml
```
Comment (kh0ma) [inline .github/workflows/dev.yml]: why do we need to specify --file pom.xml if it is a default one?

Comment (kh0ma) [inline .github/workflows/dev.yml]: Can we also attach our artifact to github packages?

Comment (alexorlenko) [reply .github/workflows/dev.yml]: ok i will clean

Comment (alexorlenko) [reply .github/workflows/dev.yml]: yes i will add

File: .github/workflows/dev.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  pull_request:
```
Comment (kh0ma) [inline .github/workflows/dev.yml:7]: On push also should be here.

The difference between pull request and push into dev should be the following:

- pull request - just build the project `mvn clean install`
- push into `dev` - upload to the Github packages - `mvn clean deploy`

Comment (kh0ma) [reply .github/workflows/dev.yml:7]: To achieve it we can either use conditions or create a different workflow.

**Review by mykytam — APPROVED** [review]

**Pull request merged**